www/artworks/post.php

```diff
@@ -9,7 +9,28 @@
 
 session_start();
 
+function post_max_size_bytes(): int{
+	$post_max_size = ini_get('post_max_size');
+	$unit = substr($post_max_size, -1);
+	$size = (int) substr($post_max_size, 0, -1);
+
+	return match ($unit) {
+		'g', 'G' => $size * 1024 * 1024 * 1024,
+		'm', 'M' => $size * 1024 * 1024,
+		'k', 'K' => $size * 1024,
+		default => $size
+	};
+}
+
 try{
+	if (empty($_POST) || empty($_FILES)){
+		if ($_SERVER['CONTENT_LENGTH'] > post_max_size_bytes()){
+			throw new \Exceptions\InvalidRequestException("Request too large (maximum " . ini_get('post_max_size') . ')');
```

Comment:
When we report the error to the user we should tell them the advertised maximum file size, i.e.:

```
ini_get('upload_max_filesize')
```